app/jobs/result_monitor_load.rb

```diff
@@ -0,0 +1,19 @@
+# Load a result from S3 into the db
+require 'logger'
+class ResultMonitorLoad
+  @queue = :q03_pipeline_run
+  @logger = Logger.new(STDOUT)
+
+  def self.perform(pipeline_run_id, load_db_command_func)
+    @logger.info("#{load_db_command_func} for pipeline run #{pipeline_run_id}")
+    pr = PipelineRun.find(pipeline_run_id)
```

Comment:
this will raise RecordNotFound if it can't find the id, so maybe this can be in the 'begin' block for better handling or you can use find_by id and check for nil

Reply:
Hmm I don't think this case can ever occur. Unless someone deleted the pipeline_run while the job was waiting in the queue. In which case I think we SHOULD see the error directly in the resque log rather than going through the rescue clause